e2e/pages/match/searchPage.ts

```diff
@@ -15,8 +15,10 @@ export class SearchPage extends MatchBasePage {
     await this.page.getByRole('button', { name: 'Update' }).click()
   }
 
-  async selectFirstAP() {
-    await this.page.getByRole('link', { name: 'View spaces' }).first().click()
+  async selectAp(premisesName: string) {
+    await this.page
+      .locator(`//h3[contains(text(),"${premisesName}")]/parent::div//a[contains(text(),"View spaces")]`)
+      .click()
```

Comment:
We don't tend to use XPath for selectors, so this may be more consistent with existing code:


```suggestion
    await this.page
      .locator('.govuk-summary-card', { has: this.page.getByRole('heading', { name: premisesName }) })
      .getByRole('link', { name: 'View spaces' })
      .click()
```